node/core/pvf/src/host.rs

```diff
@@ -416,12 +418,17 @@ async fn handle_execute_pvf(
 			},
 		}
 	} else {
-		// Artifact is unknown: register it and enqueue a job with the corresponding priority and
-		//
-		artifacts.insert_preparing(artifact_id.clone());
-		send_prepare(prepare_queue, prepare::ToQueue::Enqueue { priority, pvf }).await?;
+		if let Pvf::Preimage(inner) = pvf {
+			// Artifact is unknown: register it and enqueue a job with the corresponding priority and
+			//
+			artifacts.insert_preparing(artifact_id.clone());
+			send_prepare(prepare_queue, prepare::ToQueue::Enqueue { priority, pvf: inner }).await?;
 
-		awaiting_prepare.add(artifact_id, params, result_tx);
+			awaiting_prepare.add(artifact_id, params, result_tx);
+		} else {
+			// Expect another request with PVF provided.
+			let _ = result_tx.send(Err(ValidationError::ArtifactNotFound));
+		};
```

Comment:
Nit: `;` seems to be superfluous. I wonder why rustfmt allows that

```suggestion
		}
```